python/sglang/srt/model_executor/cuda_graph_runner.py

```diff
@@ -383,6 +417,150 @@ def __init__(self, model_runner: ModelRunner):
                 f"Capture cuda graph failed: {e}\n{CUDA_GRAPH_CAPTURE_FAILED_MSG}"
             )
 
+    def capture_prefill(self):
+        self.capture_prefill_seqlen, self.compile_prefill_seqlen = (
+            get_prefill_seqlen_to_capture(self.model_runner)
+        )
+        rank0_log(f"Capture prefill cuda graph seqlen {self.capture_prefill_seqlen}")
+        self.capture_forward_mode = ForwardMode.EXTEND
+        self.capture_hidden_mode = CaptureHiddenMode.NULL
+        # Attention backend
+        self.max_prefill_bs = 1
+        self.prefill_num_tokens_per_bs = max(self.capture_prefill_seqlen)
+        self.max_prefill_num_token = (
+            self.max_prefill_bs * self.prefill_num_tokens_per_bs
+        )
+
+        num_tokens_per_bs = 1
+        if self.model_runner.spec_algorithm.is_eagle():
+            if self.model_runner.is_draft_worker:
+                raise RuntimeError("This should not happen")
+            else:
+                num_tokens_per_bs = (
+                    self.model_runner.server_args.speculative_num_draft_tokens
+                )
+
+        # Attention backend
+        max_bs = max(self.capture_bs)
+        max_decode_num_token = max_bs * num_tokens_per_bs
+        self.max_num_token = (
+            self.max_prefill_num_token
+            if self.max_prefill_num_token > max_decode_num_token
+            else max_decode_num_token
+        )
+
+        self.model_runner.attn_backend.init_cuda_graph_state(
```

Comment:
hmm it seems we init_cuda_graph_state twice: once at GraphRunner.init, another at capture_prefill. Is it possible we only init it once at the GraphRunner.init function, since the new code looks asymmetric and the "init things twice" may cause confusion.